ignite/pkg/cosmosanalysis/app/app.go

```diff
@@ -99,24 +102,45 @@ func FindRegisteredModules(chainRoot string) ([]string, error) {
 		return nil, err
 	}
 
-	var basicModules []string
+	// The directory where the app file is located.
+	// This is required to resolve references within the app package.
+	appDir := filepath.Dir(appFilePath)
+
 	ast.Inspect(f, func(n ast.Node) bool {
-		if pkgsReg := findBasicManagerRegistrations(n, basicManagerModule); pkgsReg != nil {
-			for _, rp := range pkgsReg {
-				importModule := packages[rp]
-				basicModules = append(basicModules, importModule)
+		// Stop traversing the child nodes when there is an error
+		if err != nil {
+			return false
+		}
```

Comment:
Not sure this condition is useful, because you already have `return false` below when `err != nil`. 

Not sure I like this global `err` variable too :P

Reply:
I don't quite like it either but it's better than using `panic` and `recover`. The idea there is to make `ast.Inspect()` stop traversing nodes when there was an error in a previous inspection callback call. After an error we need to stop traversing the rest of the potential children that might be waiting to be traversed after the error happens.

Reply:
ah yes I remember now you told me about that. I didn't realize this was the reason of having this global `err`. Fine for me then!